packages/core/src/components/spinner/spinner.tsx

```diff
@@ -50,36 +50,18 @@ export class Spinner extends React.PureComponent<ISpinnerProps, {}> {
             className,
         );
 
-        const style: React.CSSProperties = {
+        const headStyle: React.CSSProperties = {
             strokeDasharray: `${PATH_LENGTH} ${PATH_LENGTH}`,
             // default to quarter-circle when indeterminate
             // IE11: CSS transitions on SVG elements are Not Supported :(
             strokeDashoffset: PATH_LENGTH - PATH_LENGTH * (value == null ? 0.25 : clamp(value, 0, 1)),
         };
 
-        // HACKHACK to squash error regarding React.SVGProps missing prop pathLength
-        const svgPathAttributes: React.DOMAttributes<SVGPathElement> = {
-            className: Classes.SPINNER_HEAD,
-            d: SPINNER_TRACK,
-            pathLength: PATH_LENGTH,
-            style,
-        } as any;
-
-        return this.renderContainer(
-            classes,
-            <svg viewBox={classes.indexOf(Classes.SMALL) >= 0 ? "-15 -15 130 130" : "0 0 100 100"}>
-                <path className={Classes.SPINNER_TRACK} d={SPINNER_TRACK} />
-                <path {...svgPathAttributes} />
-            </svg>,
-        );
-    }
-
-    // abstract away the container elements so SVGSpinner can do its own thing
-    protected renderContainer(classes: string, content: JSX.Element) {
         return (
-            <div className={classes}>
-                <div className={Classes.SPINNER_SVG_CONTAINER}>{content}</div>
-            </div>
+            <svg className={classes} viewBox="0 0 100 100">
```

Comment:
I don't fully understand SVG/viewBox and stuff, but it's amazing that you were able to dramatically simplify